analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -607,27 +607,44 @@ def main(args):
                   cls=log_parser.CompileCommandEncoder)
 
     context = analyzer_context.get_context()
-    metadata = {'action_num': len(actions),
-                'command': sys.argv,
-                'versions': {
-                    'codechecker': "{0} ({1})".format(
-                        context.package_git_tag,
-                        context.package_git_hash)},
-                'working_directory': os.getcwd(),
-                'output_path': args.output_path,
-                'result_source_files': {}}
+    metadata = {
+        'version': 2,
+        'tools': [{
+            'name': 'codechecker',
+            'action_num': len(actions),
+            'command': sys.argv,
+            'version': "{0} ({1})".format(context.package_git_tag,
+                                          context.package_git_hash),
+            'working_directory': os.getcwd(),
+            'output_path': args.output_path,
+            'result_source_files': {},
+            'analyzers': {}
+        }]}
```

Comment:
I believe the `tools` will be UBSanitizer, cppcheck and other analyzers, correct? Will the format of a tools' info be flexible? I mean for example `analyzers` attribute makes sense only in CodeChecker where we have ClangSA and ClangTidy. Will cppcheck have an `analyzers` attribute with exactly one element? Or will it be possible to eliminate this unnecessary level at other tools? Or will CodeChecker handle this logic for all tools?